src/hb-ot-shape-complex-arabic-pua.hh

```diff
@@ -0,0 +1,289 @@
+
```

Comment:
So, you hand-generated this file? Let's reuse packtab to make it much more compact. There's so many zeros in there.

I don't understand why it maps from more than just the first page though.

Reply:
I can generate a text file in a format similar to Unicode encoding mapping files (e.g. https://unicode.org/Public/MAPPINGS/VENDORS/MICSFT/WINDOWS/CP1256.TXT), would that be OK?

Reply:
In f61add0ec8ca114377d042a5d37258d76f4ac64b I committed two files using the Unicode mapping format (which mean they map from PUA to Unicode, the reverse of the C arrays).

Reply:
I did a packTab impl of it to generate table from those two files. Takes only 2kb instead of 3.5 current implementation takes. Is slower but we don't care.

Reply:
Here's the packtab gist:
https://gist.github.com/behdad/6cdb9a9a2f029b91ed8a1a6a253a5723

Reply:
Does not look like we can use `hb-ot-shaper-arabic-table.hh` here as it expects `JOINING_TYPE_*` to be defined before including it, which is why I created a new file.

Reply:
Yeah a new file is fine. Then make a new tiny generator. Which makes sense since these files are static and need no Unicode data file.

Reply:
https://github.com/harfbuzz/harfbuzz/pull/3668